sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/ParallelDocumentQueryExecutionContext.java

```diff
@@ -44,22 +46,23 @@
 public class ParallelDocumentQueryExecutionContext<T extends Resource>
         extends ParallelDocumentQueryExecutionContextBase<T> {
     private final CosmosQueryRequestOptions cosmosQueryRequestOptions;
-    private final Map<PartitionKeyRange, String> partitionKeyRangeToContinuationTokenMap;
+    private final Map<FeedRangeEpkImpl, String> partitionKeyRangeToContinuationTokenMap;
 
     private ParallelDocumentQueryExecutionContext(
-            DiagnosticsClientContext diagnosticsClientContext,
-            IDocumentQueryClient client,
-            List<PartitionKeyRange> partitionKeyRanges,
-            ResourceType resourceTypeEnum,
-            Class<T> resourceType,
-            SqlQuerySpec query,
-            CosmosQueryRequestOptions cosmosQueryRequestOptions,
-            String resourceLink,
-            String rewrittenQuery,
-            String collectionRid,
-            boolean isContinuationExpected,
-            boolean getLazyFeedResponse,
-            UUID correlatedActivityId) {
+        DiagnosticsClientContext diagnosticsClientContext,
+        IDocumentQueryClient client,
+        List<PartitionKeyRange> partitionKeyRanges,
+        ResourceType resourceTypeEnum,
+        Class<T> resourceType,
+        SqlQuerySpec query,
+        CosmosQueryRequestOptions cosmosQueryRequestOptions,
+        String resourceLink,
+        String rewrittenQuery,
+        String collectionRid,
+        boolean isContinuationExpected,
+        boolean getLazyFeedResponse,
+        UUID correlatedActivityId,
+        List<FeedRangeEpkImpl> feedRanges) {
```

Comment:
>List [](start = 8, length = 4)

These should be immutable lists so that the caller knows they won't get mutated.

Reply:
I dont need these anymore. Removed